0000-py-version-support.md

```diff
@@ -0,0 +1,244 @@
+|     31     |    TBD (possibly oct 2020)  |      3.7        |
+|     32     |    TBD (possibly apr 2020)  |      3.8        |
+
+Based off of these dates and Python's End of Life dates, Fedora will not
+be of any concern with this change in policy. Fedora releases fairly regularly
+and includes the latest Python version. Both packages and pip installs
+will work against supported Fedora versions going forward.
+
+SLES/OpenSuse
+----
+The General End of Life dates do conflict with the current Python Version
+End of Life dates. Since Suse does the packaging here they have been informed
+of this policy change and will include a resolution around this with their packages
+when these version conflicts occur.
+
+Other Linux
+-----------
+
+|     OS     |    End of Life      | Default Python |
+|------------|---------------------|----------------|
+|     Rhel 6 |    30 Nov 2020      |      2.6       |
+|     Rhel 7 |    30 Jun 2024      |      2.7       |
+|     Rhel 8 |    05 May 2029      |      None      |
+|     Ubuntu 16.04  |    April 2021    |    2.7/3.5 |
+|     Ubuntu 18.04  |    April 2023    |    3.6     |
+|     Ubuntu 20.04  |    2030          |    3.8     |
+|     Debian 8      |    30 June 2020  |  2.7, 3.4  |
+|     Debian 9      |    June 2022     |  2.7, 3.5  |
+|     Debian 10      |    June 2024    |  2.7, 3.7  |
+
+Based on this chart above all of these Operating Systems will or already have conflicted with
+the Python's End of Life Support dates. We will need to include Python in our packages for these
+Operating Systems especially at any given time when their default Python version is considered
+End of Life. This does impact users who would want to pip install Salt. They will need to install
+a newer version of Python before they can install Salt.
+
+Other Reasonable-Effor Support Linux
+------------------------------------
+
+Arch
+----
+Arch is a rolling release and is consistantly updating their Python version, so this is not
+of any concern for this policy.
+
+FreeBSD
+-------
+According to this https://wiki.freebsd.org/Python/PortsPolicy FreeBSD currently already aligns
+with this policy. They only support Python versions until they are End of Life so there are
+no concerns with this operating system.
+
+SmartOS
+-------
+SmartOS includes the latest Python version so will not be impacted by this change.
+
+
+What about dependencies?
+------------------------
+Before dropping an End of Life Python version we should have an idea beforehand if Salt's dependencies
+do not support a particular Python version with our testing suite. Currently our tests use the dependencies
+in the requirements directory in the main Salt project. This does of course mean that if we are not testing
+a module we would not have this information beforehand. We are now requiring tests to be added
+for PRs so this will help ensure this is caught for those modules, but there are still some modules that do
+not have tests currently.
```

Comment:
should this not be addressed when a change of library version in the requirements.txt goes to the CI under the minimal Python version supported?

Reply:
We do not have all libraries for all salt modules in the requirements file, only ones that have test written for them. And just because a library would install on a particular python version does not mean it fully supports that version. Thats why i had to do the extra step to check the extra libraries by installing the requisites to check version support.